CMakeLists.txt

```diff
@@ -120,16 +120,18 @@ ENDIF()
 # These are the variables we will append to as we go
 # I really wish these were regular variables
 # but scoping issues can make it difficult
-GLOBAL_SET(KOKKOS_COMPILE_OPTIONS)
-GLOBAL_SET(KOKKOS_COMPILE_DEFINITIONS)
-GLOBAL_SET(KOKKOS_LINK_OPTIONS)
-GLOBAL_SET(KOKKOS_LINK_LIBRARIES)
-GLOBAL_SET(KOKKOS_CUDA_OPTIONS)
-GLOBAL_SET(KOKKOS_CUDAFE_OPTIONS)
-GLOBAL_SET(KOKKOS_XCOMPILER_OPTIONS)
```

Comment:
So the only real change here is not to reset `KOKKOS_COMPILE_DEFINITIONS` and `KOKKOS_LINK_LIBRARIES`. AFAICT, all these variables are normally empty anyway.

Reply:
@jjwilke - can you address the questions about the changes?